lib/msf/core/rhosts_walker.rb

```diff
@@ -296,6 +297,27 @@ def parse_ssh_uri(value, datastore)
       result
     end
 
+    # Parses a uri string such as tcp://user:password@example.com into a hash
+    # which can safely be merged with a [Msf::DataStore] datastore for setting options.
+    #
+    # @param value [String] the uri string
+    # @return [Hash] A hash where keys match the required datastore options associated with
+    #   the uri value
+    def parse_tcp_uri(value, datastore)
+      uri = ::Addressable::URI.parse(value)
```

Comment:
Should this not have exception handling here?

Reply:
There's exception handling on the caller's side :+1:

Reply:
Sounds good, marking resolved.